### ISSUE TITLE
[cvp 2019]change control shows failed but the intended changes are applied according to the playbook

### ISSUE THREAD

**Issue by roman-so** [aristanetworks/ansible-cvp#140]
Issue Type
------

 - Question / Bug Report

Module Name
------
- cv_device
- cv_task

`arista.cvp` collection and Python libraries version

```
ansible 2.9.2
  config file = /scripts/ansible.cfg
  configured module search path = [u'/plugins/modules', u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.15 (default, Jan 31 2019, 00:15:21) [GCC 6.3.0]

DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
ansible==2.9.2
certifi==2019.11.28
cffi==1.13.2
chardet==3.0.4
cryptography==2.8
enum34==1.1.6
idna==2.8
ipaddress==1.0.23
Jinja2==2.10.3
MarkupSafe==1.1.1
pycparser==2.19
PyYAML==5.2
requests==2.22.0
six==1.13.0
treelib==1.5.5
urllib3==1.25.7


```

OS / Environment
------
Build Information
CloudVision version
2019.1.1
UI version
1.13.0
Build hash
74033b2f7
Build time
Nov 20, 2019 17:46:23 CET

Summary
------

Using the below playbook the intended changes are successefully applied but the change control shows  
> CC completed with error: Error executing stage Task_0_22 : Task 22 failed
![image](https://user-images.githubusercontent.com/59204786/74760039-13ce5b00-527a-11ea-89aa-08b1b5db2777.png)

**and it takes 900second each time for Terminattr update** 

Steps to reproduce
------
**The playbook**

```
- name: Test cv_device
  hosts: cvp
  connection: local
  gather_facts: no
  collections:
    - arista.cvp
  
  vars:
    configlet_list:
      cv_device_test01: "{{ lookup('file', 'configlets/configlet_cv_device_test01.txt') }}"
      cv_device_test02: "{{ lookup('file', 'configlets/configlet_cv_device_test02.txt') }}"
      BASE_108: "{{ lookup('file', 'configlets/BASE_108.txt') }}"      
    configlet_filter_list:
      - cv_device_test01
      #- cv_device_test02      
      - BASE_108
      - SYS_TelemetryBuilderV3
      - SYS_TelemetryBuilderV3_x.x.x.x_1

    # Device inventory for provision activity: bind configlet
    device_filter_list:
        - veos8
    devices_inventory:
      veos8: # ce nom peut ne pas correspondre au nom de l'équipment
        name: veos8
        configlets:
          - cv_device_test01
          #- cv_device_test02
          - BASE_108
          - SYS_TelemetryBuilderV3
          - SYS_TelemetryBuilderV3_x.x.x.x_1
        parent_container: IPFabric1 

  tasks:

###################################################
### Configlets deloyment
###################################################

    # Collect CVP Facts before cv_configlets
    - name: "Gather CVP facts from {{inventory_hostname}}"
      arista.cvp.cv_facts:
      register: cvp_facts

    - name: 'Create configlets on CVP {{inventory_hostname}}.'
      cv_configlet:
        cvp_facts: "{{ cvp_facts.ansible_facts }}"
        configlets: "{{ configlet_list }}"
        configlet_filter: "{{ configlet_filter_list }}"
        state: present
      register: CVP_CONFIGLET_RESULT

    - name: Display cvp_configlet 
      debug:
        msg: "{{ CVP_CONFIGLET_RESULT }}"

    - name: "Executer les tâches en suspends avec un delai d'attente de 60 sec"
      arista.cvp.cv_task:
        tasks: "{{ CVP_CONFIGLET_RESULT.data.tasks }}"   
        wait: 60


###################################################
### Devices deployment
###################################################

      # Collect CVP Facts before cv_devices
    - name: "Gather CVP facts from {{inventory_hostname}}"
      arista.cvp.cv_facts:
      register: cvp_facts

    # Display information for device before a container's move
    - name: "Bind configlets to devices on {{ inventory_hostname }}"
      cv_device:
        devices: "{{ devices_inventory }}"
        cvp_facts: "{{ cvp_facts.ansible_facts }}"
        device_filter: "{{ device_filter_list }}"
        state: present
      register: CVP_DEVICE_RESULT

    - name: Display cvp_configlet 
      debug:
        msg: "{{ CVP_DEVICE_RESULT }}"

    - name: "Executer les tâches en suspends avec un delai d'attente de 60 sec"
      arista.cvp.cv_task:
        tasks: "{{ CVP_DEVICE_RESULT.data.tasks }}"   
        wait: 60
```
**The result:**

> py2.7-ansible-cvp:root% ansible-playbook  playbook.device.configlet.2019.yaml
> [DEPRECATION WARNING]: The TRANSFORM_INVALID_GROUP_CHARS settings is set to allow bad characters in group names by default, 
> this will change, but still be user configurable on deprecation. This feature will be removed in version 2.10. Deprecation 
> warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
> [WARNING]: Invalid characters were found in group names but not replaced, use -vvvv to see details
> 
> 
> PLAY [Test cv_device] *********************************************************************************************************
> 
> TASK [Gather CVP facts from cvp_2019] *****************************************************************************************
> Tuesday 18 February 2020  16:14:46 +0000 (0:00:00.130)       0:00:00.130 ****** 
> ok: [cvp_2019]
> 
> TASK [Create configlets on CVP cvp_2019.] *************************************************************************************
> Tuesday 18 February 2020  16:14:53 +0000 (0:00:06.643)       0:00:06.774 ****** 
> changed: [cvp_2019]
> 
> TASK [Display cvp_configlet] **************************************************************************************************
> Tuesday 18 February 2020  16:15:06 +0000 (0:00:13.052)       0:00:19.827 ****** 
> ok: [cvp_2019] => {
>     "msg": {
>         "changed": true, 
>         "data": {
>             "deleted": [], 
>             "new": [
>                 {
>                     "cv_device_test02": "success"
>                 }, 
>                 {
>                     "cv_device_test01": "success"
>                 }
>             ], 
>             "tasks": [], 
>             "updated": []
>         }, 
>         "failed": false
>     }
> }
> 
> TASK [Executer les tâches en suspends avec un delai d'attente de 60 sec] ******************************************************
> Tuesday 18 February 2020  16:15:07 +0000 (0:00:00.959)       0:00:20.787 ****** 
> [WARNING]: No actionable tasks found on CVP
> 
> ok: [cvp_2019]
> 
> TASK [Gather CVP facts from cvp_2019] *****************************************************************************************
> Tuesday 18 February 2020  16:15:08 +0000 (0:00:01.536)       0:00:22.323 ****** 
> ok: [cvp_2019]
> 
> TASK [Bind configlets to devices on cvp_2019] *********************************************************************************
> Tuesday 18 February 2020  16:15:15 +0000 (0:00:06.676)       0:00:29.000 ****** 
> changed: [cvp_2019]
> 
> TASK [Display cvp_configlet] **************************************************************************************************
> Tuesday 18 February 2020  16:15:32 +0000 (0:00:16.442)       0:00:45.443 ****** 
> ok: [cvp_2019] => {
>     "msg": {
>         "changed": true, 
>         "data": {
>             "new": [], 
>             "reset": [], 
>             "tasks": [
>                 {
>                     "currentTaskName": "Submit", 
>                     "description": "Ansible Configlet Update: on Device veos8", 
>                     "note": "", 
>                     "taskStatus": "ACTIVE", 
>                     "workOrderDetails": {
>                         "ipAddress": "10.157.121.108", 
>                         "netElementHostName": "veos8", 
>                         "netElementId": "50:00:00:03:37:66", 
>                         "serialNumber": "14EF18BDF8B48402440B24349E1FA911", 
>                         "workOrderDetailsId": "", 
>                         "workOrderId": ""
>                     }, 
>                     "workOrderId": "22", 
>                     "workOrderState": "ACTIVE", 
>                     "workOrderUserDefinedStatus": "Pending"
>                 }
>             ], 
>             "updated": [
>                 {
>                     "veos8": "Configlets-[u'22']"
>                 }
>             ]
>         }, 
>         "failed": false
>     }
> }
> 
> TASK [Executer les tâches en suspends avec un delai d'attente de 60 sec] ******************************************************
> Tuesday 18 February 2020  16:15:33 +0000 (0:00:00.954)       0:00:46.398 ****** 
> [WARNING]: Task 22 has not completed in 60 seconds
> 
> changed: [cvp_2019]
> 
> PLAY RECAP ********************************************************************************************************************
> cvp_2019                   : ok=8    changed=3    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
> 
> Tuesday 18 February 2020  16:16:35 +0000 (0:01:02.541)       0:01:48.939 ****** 
> 
> =============================================================================== 
> Executer les tâches en suspends avec un delai d'attente de 60 sec ----------------------------------------------------- 62.54s
> Bind configlets to devices on cvp_2019 -------------------------------------------------------------------------------- 16.44s
> Create configlets on CVP cvp_2019. ------------------------------------------------------------------------------------ 13.05s
> Gather CVP facts from cvp_2019 ----------------------------------------------------------------------------------------- 6.68s
> Gather CVP facts from cvp_2019 ----------------------------------------------------------------------------------------- 6.64s
> Executer les tâches en suspends avec un delai d'attente de 60 sec ------------------------------------------------------ 1.54s
> Display cvp_configlet -------------------------------------------------------------------------------------------------- 0.96s
> Display cvp_configlet -------------------------------------------------------------------------------------------------- 0.95s
> py2.7-ansible-cvp:root% 

**The logs:**
CVP Arista change control logs

[cc_CC_Task_fee61a02-39d3-4a68-8b3e-c07ff2b0d1b1_02182020165000.log](https://github.com/aristanetworks/ansible-cvp/files/4220679/cc_CC_Task_fee61a02-39d3-4a68-8b3e-c07ff2b0d1b1_02182020165000.log)

**Expected results**

```
Successefully executed with no errors (the same playbook runs successfully without any errors in cvp 2018)
```
Actual results
------

[cvp 2019]change control shows failed but the intended changes are applied according to the playbook and it takes 900second each time for Terminattr update 

> Update Config
> veos8
> Waiting up to 900 seconds for Terminattr update from: 14EF18BDF8B48402440B24349E1FA911

**Comment by titom73** [comment]
Hello @roman-so and thank you for reporting issue.

Have you pull out latest version of the collection ? Indeed, I see you are running with a python 2.7 environment where `arista.cvp` collection has moved to full Python3 support.

Could you check version you are running in your environment:

```
$ cat ansible_collections/arista/cvp/galaxy.yml| grep version:
$ git show
```

Thanks

**Comment by AJNOURI** [comment]
Hi @titom73  and thank you for your reply.
With @roman-so , we're from the same team.

indeed, the docker environment, we are using is a built container from "Dockerfile-2.7" and not the latest

> py2.7-ansible-cvp:root% cat /project/arista/cvp/galaxy.yml | grep version:
> version: 1.0.2
> 

Outside of the container:

> $ git version
> git version 2.7.4

So, do you recommend using "Dockerfile-3" for the docker environment?

----------

We will do the same test with the latest "Dockerfile-3" and get back to you.

**Comment by titom73** [comment]
Hello @AJNOURI 

Yes please try with a Py3 environment.

Besides that, message comes from CloudVision side meaning that you should investigate on that side to see if using manual action raise same behavior: `Waiting up to 900 seconds for Terminattr update from ...`

Please keep us posted about results.

**Comment by AJNOURI** [comment]
Either manually or through Ansible-CVP, it is the same behavior.

>  Waiting up to 900 seconds for Terminattr update from ...

---
We've tried to build the image from Dockerfile-3, but we got the below error with libssl1.1:

> Step 10/17 : RUN apk add --update --no-cache ca-certificates                                 openssh-client                                 build-base                                 gcc                                 g++                                 make                                 python-dev                                 py-pip                                 libffi-dev                                 sshpass                                 libssl1.1                                 openssl-dev &&                                pip install --upgrade pip &&                                 pip install -r requirements.txt &&                                apk del -r --purge gcc make g++ &&                                rm -rf /source/* &&                                rm -rf /var/cache/apk/* &&                                rm -rf /tmp/*
>  ---> Running in a0f3c7c2e2b6
> fetch http://dl-cdn.alpinelinux.org/alpine/v3.6/main/x86_64/APKINDEX.tar.gz
> fetch http://dl-cdn.alpinelinux.org/alpine/v3.6/main/x86_64/APKINDEX.tar.gz
> fetch http://dl-cdn.alpinelinux.org/alpine/v3.6/community/x86_64/APKINDEX.tar.gz
> fetch http://dl-cdn.alpinelinux.org/alpine/v3.6/community/x86_64/APKINDEX.tar.gz
> ERROR: unsatisfiable constraints:
>   libssl1.1 (missing):
>     required by: world[libssl1.1]
> 

So, we've replaced libssl1.1 by libssl1.0, and the build succeeded.
Should we stick with libssl1.0 ?

**Comment by titom73** [comment]
Thanks for the feedback.

I would recommend to first work on CloudVision to find the reason you have this issue and then we will re-check with the collection.

For Docker issue, it is already updated in the next release branch with [this commit](https://github.com/aristanetworks/ansible-cvp/commit/fc1f6a3ce32f344ef4620b256cd59e94e3c4f93e)

**Comment by roman-so** [comment]
Thank you for your assistance.
We buit the new docker image successfully.
We tested the new image with the installed cvp2019.1.1 version and the behavior was exactly the same as described above. We didn't find how to solve the issue
> Waiting up to 900 seconds for Terminattr update from:

We have installed the latest version of cvp -2019.1.3 and it worked with the playbook mentioned above.

Thank you @titom73
